generator/build_integration.zig

```diff
@@ -43,11 +47,12 @@ pub const ShaderCompileStep = struct {
     /// This path can then be used to include the binary into an executable, for example by passing it
     /// to @embedFile via an additional generated file.
     pub fn add(self: *ShaderCompileStep, src: []const u8) []const u8 {
+        const output_filename = std.fmt.allocPrint(self.builder.allocator, "{s}.spv", .{ src }) catch unreachable;
```

Comment:
Not sure this is the best way to do this - don't have a lot of experience with Zig string manipulation.

Reply:
This seems fairly reasonable to me. What would you otherwise suggest?

Reply:
`src` is a relative path to the source directory, and the shader's output binary is placed in the same path but relative to the cache directory so that names don't collide. So this seems correct to me.